File: leetcode/80.remove-duplicates-from-sorted-array-2.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        
        top = 1

        if len(nums) == 1:
            return top

        for i in range(top+1, len(nums)):
            if nums[top-1] != nums[i]:
                nums[top+1] = nums[i]
                top+=1

        top+=1
        return top;        

File: leetcode/80.remove-duplicates-from-sorted-array-2.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        
        if( nums.size() == 1)
            return 1;
            
        int top = 1;



        for( int i = top +1; i< nums.size(); i++)
        {
            if( nums[top -1] != nums[i])
            {
                nums[top+1] = nums[i];
                top++;
            }
        }

        return ++top;
    }
};